pkg/migrate/azure/dns.go

```diff
@@ -0,0 +1,1335 @@
+		}
+		if rs.ARecords != nil && len(*rs.ARecords) > 0 {
+			showLegacyARecords(rs.ARecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.AaaaRecords != nil && len(*rs.AaaaRecords) > 0 {
+			showLegacyAaaaRecords(rs.AaaaRecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.PtrRecords != nil && len(*rs.PtrRecords) > 0 {
+			showLegacyPtrRecords(rs.PtrRecords, zoneName, *rs.Fqdn, ttl)
+		}
+	}
+}
+
+// DoListLegacyZone Retrives legacy zone info and prints it out
+func DoListLegacyZone(resourceGroup string, legacyZone string, public bool) error {
+	session, err := azconfig.GetSession()
+	if err != nil {
+		return err
+	}
+
+	legacyDNSClient := newLegacyDNSClient(session, resourceGroup, public)
+
+	// Serialize/Deserialize to make deep copy
+	szone, err := legacyDNSClient.GetZoneSerialized(legacyZone)
+	if err != nil {
+		return err
+	}
+
+	legacyDNSZoneInfo, err := deserializeLegacyZone(szone)
+	if err != nil {
+		return err
+	}
+
+	doListLegacyZoneDeserialized(legacyDNSZoneInfo)
+	return nil
+}
+
+// Prints out a single private zone
+func doListPrivateZoneDeserialized(privateDNSZoneInfo *privateDNSZoneInfo) {
+	logrus.Debug(spew.Sdump(privateDNSZoneInfo))
+
+	zoneName := safeString(privateDNSZoneInfo.zone.Name)
+
+	logrus.Printf("PrivateZone: %s", zoneName)
+	for _, rs := range privateDNSZoneInfo.recordsets {
+		ttl := int64(30)
+		if rs.TTL != nil {
+			ttl = *rs.TTL
+		}
+
+		if rs.SoaRecord != nil {
+			showPrivateSoaRecord(rs.SoaRecord, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.MxRecords != nil && len(*rs.MxRecords) > 0 {
+			showPrivateMxRecords(rs.MxRecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.CnameRecord != nil {
+			showPrivateCnameRecord(rs.CnameRecord, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.SrvRecords != nil && len(*rs.SrvRecords) > 0 {
+			showPrivateSrvRecords(rs.SrvRecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.TxtRecords != nil && len(*rs.TxtRecords) > 0 {
+			showPrivateTxtRecords(rs.TxtRecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.ARecords != nil && len(*rs.ARecords) > 0 {
+			showPrivateARecords(rs.ARecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.AaaaRecords != nil && len(*rs.AaaaRecords) > 0 {
+			showPrivateAaaaRecords(rs.AaaaRecords, zoneName, *rs.Fqdn, ttl)
+		}
+		if rs.PtrRecords != nil && len(*rs.PtrRecords) > 0 {
+			showPrivatePtrRecords(rs.PtrRecords, zoneName, *rs.Fqdn, ttl)
+		}
+	}
+
+}
+
+// DoListPrivateZone Retrives private zone info and prints it out
+func DoListPrivateZone(resourceGroup string, privateZone string) error {
+	session, err := azconfig.GetSession()
+	if err != nil {
+		return err
+	}
+
+	privateDNSClient := newPrivateDNSClient(session, resourceGroup, "", "")
+
+	// Serialize/Deserialize to make deep copy
+	szone, err := privateDNSClient.GetZoneSerialized(privateZone)
+	if err != nil {
+		return err
+	}
+
+	privateDNSZoneInfo, err := deserializePrivateZone(szone)
+	if err != nil {
+		return err
+	}
+
+	doListPrivateZoneDeserialized(privateDNSZoneInfo)
```

Comment:
can you explain a bit why there is a requirement for deepcopy? and why are we serializing de serializing? also why aren't we using json in that case, instead of `gob` ?

Reply:
I've commented the code here for the reasons why it was done this way.

Reply:
@jhixson74 I can't see where you added the comment, can you point me to it in new changset?

Also can you summarize why here too if possible?

Reply:
It got nuked when this section of code got nuked ;-) Here: https://github.com/openshift/installer/commit/50477870a9fa434e354b6bb39f4d8bf53f932e6f#diff-4727bf8fc830a99ae17aaee2bf9f0021R1084

Reply:
```
	// Serialize/Deserialize to make deep copy
	// It is necessary to make a deep copy since there are several
	// references in various structures that go out of scope, I assume they
	// are garbage collected. The necessity to use gob is that most of
	// these structures have JSON marshal/unmarshal methods that do not
	// include necessary properties that are needed here.
```

just for discussion sake.

go shouldn't drop memory that's still in use. so i don't think this serialize/serialize is necessary.. 
can you provide exact example of the error you are seeing or drop these please.

Reply:
I can provide an example, but it's no longer necessary since I have nuked most of this code from orbit ;-).